compiler/z/codegen/OpMemToMem.cpp

```diff
@@ -1556,32 +1557,24 @@ MemCmpConstLenMacroOp::generate(TR::Register* dstReg, TR::Register* srcReg, TR::
    _litReg = NULL;
    TR::Compilation *comp = _cg->comp();
 
+   if (_length == 0)
```

Comment:
if the length is zero then arrays are equal and we should return zero in result register. No other instructions needed.